src/diffcalc_API/services/hkl.py

```diff
@@ -156,3 +215,36 @@ async def calculate_ub(
 
     await store.save(name, hklcalc, collection)
     return np.round(hklcalc.ubcalc.UB, 6).tolist()
+
+
+@async_wrap
```

Comment:
Does this function need to be async? It doesn't seem to await anything? Ditto for `combine_lab_position_results()` and `generate_axis()`

Reply:
Yeah, I added the async_wrap functionality because I read that it's bad practise to mix async and sync code. So I guess, just incase these calls become intensive (which is a possibility), making them execute asynchronously seemed sensible. For example, a person could ask for very high resolution scans, at which point `combine_lab_position_results` becomes blocking.

Reply:
Making a function async doesn't automatically prevent it from blocking. These functions don't contain any `await`s, so they will block regardless. `await` releases control of an async function to the event loop, allowing it to interleave operations around `await` statements and provide some concurrency, but if you're not touching async code, it may not be worth it.

In this case, `combine_lab_position_results` will block regardless, because the event loop will just call it, naively assuming that it will either run very quickly or contain `await`s. 

I usually just assume that `await`less code is okay, although there are exceptions. If `combine_lab_position_results` starts taking too long, we can address that, but async code isn't the best method, it's better for interleaving I/O.

Reply:
I tend to just assume that code without `await`s isn't blocking until proven otherwise

Reply:
OK, fair enough. I'll get rid of this for now.